### PR TITLE
Feat / refactor containers

### PR DESCRIPTION
**Pull request by royschut**
### Goal
- Move logic from containers to controllers as much as possible, so that it can be reused from all implementations

### Solution
I've refactored the containers as followed:
- Moved as much of the logics to the controllers as possible
- Improved useForm to await the submit handler, so that it can implement the api errors
- Created a new FormValidationError class for such errors
- Moved all controller calls into useMutation, so that it can handle the loading state and errors for us. This makes the try/catch blocks unnecessary and hopefully makes the code more readable
- Added a new useCheckout hook, so that the logic implementations can potentially also be used by react-native

### Notes
- The Checkout container especially has been refactored a lot, so it is kinda scary. However:
- I've tested all payment methods, but Cleeng Adyen currently gives a "Payment refused" error (which also happens on our [demo web app](https://svod-cleeng-acc.demo.videodock.com/?u=checkout)).
- I still have to refactor ChooseOffer, but decided to do this separate, as it becomes such a large PR. There are also a lot of related components/containers that could/should be refactored (AdyenPaymentInitial, AdyenPaymentDetails, UpdatePaymentDetails), but I had to draw the line somewhere.

### CODE REVIEW

**Comment by royschut** [comment]
I see the merge conflicts, but this is a large one which I will resolve later.

File: packages/ui-react/src/components/PaymentForm/PaymentForm.tsx [hunk 1]
```diff
@@ -1,121 +1,108 @@
...
-      <div>
-        <CreditCardNumberField
-          value={paymentData?.values?.cardNumber?.toString()}
-          error={paymentData?.errors?.cardNumber}
-          onBlur={paymentData?.handleBlur}
-          onChange={paymentData?.handleChange}
-        />
-      </div>
-      <div className={styles.columns}>
+      <form onSubmit={handleSubmit} data-testid={testId('payment_form')} noValidate>
```
Comment (langemike) [inline packages/ui-react/src/components/PaymentForm/PaymentForm.tsx:74]: I'm a bit suprised to see this change in this PR. Can you elaborate?

**Review by langemike — APPROVED** [review]
Based on a quickscan I approve. Since melissa is not here this week, I hope we can merge this before friday with 2 approvals. I only have one code question (see inline comment)...

File: packages/ui-react/src/components/PaymentForm/PaymentForm.tsx (hunk 1 above)
Comment (royschut) [reply packages/ui-react/src/components/PaymentForm/PaymentForm.tsx:74]: This component was using `useForm`, but not a `<form />` element with a submit handler. Also the error wasn't presented anywhere.

**Review by ChristiaanScheermeijer — APPROVED** [review]
PR looks good! We know that it is not ready, but to prevent further merge conflicts, we agreed to combine our work on the feat/restructure-sprint-2 branch.

**Pull request merged**